### PR TITLE
[FEATURE] - Kawaii ipsec

### PR DESCRIPTION
**Pull request by XT-Phate**

### CODE REVIEW

File: openapi/fragments/kawaii_ipsec.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{{ oas.StringProperty('pre_shared_key', 'The ' + resources.kawaii_ipsec.desc + ' pre-shared key(PSK)' )}}
+{{ oas.StringProperty('dpd_timeout_action', 'The ' + resources.kawaii_ipsec.desc + ' Dead Peer Detection Action (clear,restart or trap)', ['clear,restart, start'], 'restart' )}}
+{{ oas.StringProperty('dpd_timeout', 'The ' + resources.kawaii_ipsec.desc + ' Dead Peer Detection Timeout', [], '240s')}}
+{{ oas.StringProperty('start_action', 'The ' + resources.kawaii_ipsec.desc + ' start action (none, start, trap)', ['none', 'start', 'trap'], 'start' )}}
+{{ oas.StringProperty('rekey_time', 'The ' + resources.kawaii_ipsec.desc + ' rekey time. Default is 2h', [], '2h' )}}
+{{ oas.StringProperty('phase1_lifetime', 'The ' + resources.kawaii_ipsec.desc + ' Lifetime for phase 1 negociation. Default is 1h', [], '1h' )}}
+{{ oas.IntProperty('phase1_dh_group_number', 'The ' + resources.kawaii_ipsec.desc + ' phase 1 Diffie Hellman IANA algorithm', [2, 5, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24] )}}
+{{ oas.StringProperty('phase1_integrity_algorithm', 'The ' + resources.kawaii_ipsec.desc + ' phase 1 integrity algorithm.', ['SHA1', 'SHA256', 'SHA384', 'SHA512'] )}}
```
Comment (gxben) [inline openapi/fragments/kawaii_ipsec.yml.j2:16]: same here

File: openapi/fragments/kawaii_ipsec.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{{ oas.StringProperty('phase1_encryption_algorithm', 'The ' + resources.kawaii_ipsec.desc + ' phase 1 encryption algorithm.', ['AES128', 'AES256', 'CAMELLIA128', 'CAMELLIA256'] )}}
```
Comment (gxben) [inline openapi/fragments/kawaii_ipsec.yml.j2:17]: and here

File: openapi/fragments/kawaii_ipsec.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{{ oas.StringProperty('phase2_lifetime', 'The ' + resources.kawaii_ipsec.desc + ' Lifetime for phase 2 negociation. Default is 1h', [], '1h' )}}
+{{ oas.IntProperty('phase2_dh_group_number', 'The ' + resources.kawaii_ipsec.desc + ' phase 2 Diffie Hellman IANA algorithm', [2, 5, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24] )}}
```
Comment (gxben) [inline openapi/fragments/kawaii_ipsec.yml.j2:19]: and here

File: openapi/fragments/kawaii_ipsec.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+{{ oas.StringProperty('phase2_integrity_algorithm', 'The ' + resources.kawaii_ipsec.desc + ' phase 2 integrity algorithm.', ['SHA1', 'SHA256', 'SHA384', 'SHA512'] )}}
```
Comment (gxben) [inline openapi/fragments/kawaii_ipsec.yml.j2:20]: ditto ;-)

File: openapi/fragments/kawaii_ipsec.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+{{ oas.StringProperty('phase2_encryption_algorithm', 'The ' + resources.kawaii_ipsec.desc + ' phase 2 encryption algorithm.', ['AES128', 'AES256', 'CAMELLIA128', 'CAMELLIA256'] )}}
```
Comment (gxben) [inline openapi/fragments/kawaii_ipsec.yml.j2:21]: and again ... which makes me think that pahse 1 and 2 params are the saame and could have been a dedicated object for mutualization

File: openapi/openapi.generated.yml [hunk 6]
```diff
@@ -396,6 +396,11 @@ components:
           description: The Kawaii list of Kowabunga private VPC subnet peering entries.
           items:
             $ref: '#/components/schemas/KawaiiVpcPeering'
+        ipsec_connections:
+          type: array
+          description: The Kawaii list of Kowabunga ipsecs connections.
```
Comment (gxben) [inline openapi/openapi.generated.yml:401]: IPSec

File: openapi/fragments/kawaii_ipsec.yml.j2 (hunk 2 above)
Comment (XT-Phate) [reply openapi/fragments/kawaii_ipsec.yml.j2:17]: mandatory

File: openapi/fragments/kawaii_ipsec.yml.j2 (hunk 3 above)
Comment (XT-Phate) [reply openapi/fragments/kawaii_ipsec.yml.j2:19]: mandatory

File: openapi/fragments/kawaii_ipsec.yml.j2 (hunk 4 above)
Comment (XT-Phate) [reply openapi/fragments/kawaii_ipsec.yml.j2:20]: mandatory

File: openapi/fragments/kawaii_ipsec.yml.j2 (hunk 5 above)
Comment (XT-Phate) [reply openapi/fragments/kawaii_ipsec.yml.j2:21]: For now, they are the same, but there could be additional configurations settings on phase2 or 1 that are different. Hence no object there for the moment

**Pull request merged**